source/common/init/manager_impl.cc

```diff
@@ -51,18 +54,34 @@ void ManagerImpl::initialize(const Watcher& watcher) {
 
     // Attempt to initialize each target. If a target is unavailable, treat it as though it
     // completed immediately.
+    uint32_t target_index = 0;
     for (const auto& target_handle : target_handles_) {
       if (!target_handle->initialize(watcher_)) {
-        onTargetReady();
+        onTargetReady(target_names_[target_index]);
       }
+      target_index++;
     }
   }
 }
 
-void ManagerImpl::onTargetReady() {
+// For init manager to query unready targets
+void ManagerImpl::checkUnreadyTargets() {
```

Comment:
This doesn't seem to be called from anywhere other than tests. Where do you expect to call this from?

Reply:
Eventually the dumped info should be returned and will be driven by admin handler when iterator over init manager owners.
https://github.com/envoyproxy/envoy/issues/11963

I think checkUnreadyTargets() should be erased or for debug only.